extractor/filesystem/containers/k8simage/k8simage.go

```diff
@@ -0,0 +1,273 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package k8simage extracts container image references from Kubernetes YAML files.
+package k8simage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem"
+	"github.com/google/osv-scalibr/extractor/filesystem/internal/units"
+	"github.com/google/osv-scalibr/inventory"
+	"github.com/google/osv-scalibr/log"
+	"github.com/google/osv-scalibr/plugin"
+	"github.com/google/osv-scalibr/purl"
+	"github.com/google/osv-scalibr/stats"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	// Name is the unique name of this extractor.
+	Name = "containers/k8simage"
+
+	// DefaultMaxFileSizeBytes is the default maximum file size the extractor will
+	// attempt to process. If a file is encountered that is larger than this
+	// limit, the file is skipped during processing.
+	DefaultMaxFileSizeBytes = 1 * units.MiB
+)
+
+// K8sResource represents a Kubernetes resource with the fields needed for image extraction.
+type K8sResource struct {
```

Comment:
Since these structs are only used in this package they can be private (i.e. start with lowercase)
Same for the ones below.